### PR TITLE
Heightmap: replace Point struct with TilePos2 vector

### PR DESCRIPTION
**Pull request by ShuraxN**
Changed point Structs to TilePos-Vectors as wanted.

### CODE REVIEW

File: src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp [hunk 1]
```diff
@@ -502,7 +502,7 @@ namespace OpenLoco::World::MapGenerator
             const auto randDensity = (gPrng1().randNext(15) * std::max(options.maxForestDensity - options.minForestDensity, 0)) / 15 + options.minForestDensity;
             placeTreeCluster(randLoc, randRadius, randDensity, std::nullopt);
 
-            if (TileManager::numFreeElements() < 0x1B000)
+            if (TileManager::numFreeElements() < 0x36000)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp:505]: Please revert this change.

Comment (ShuraxN) [reply src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp:505]: Ok I will do so.

**Review by AaronVanGeffen — APPROVED** [review]
